.pre-commit-config.yaml

```diff
@@ -1,20 +1,30 @@
-# See https://pre-commit.com for more information
-# See https://pre-commit.com/hooks.html for more hooks
 repos:
--   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.2.0
+  - repo: https://github.com/markdownlint/markdownlint
+    rev: v0.12.0
     hooks:
-    -   id: trailing-whitespace
-    -   id: end-of-file-fixer
-    -   id: check-yaml
-    -   id: check-xml
-    -   id: check-json
-    -   id: pretty-format-json
-        args: [--no-ensure-ascii, --no-sort-keys, --autofix]
-        files: ".+json"
-    -   id: check-added-large-files
-# - repo: https://github.com/jumanjihouse/pre-commit-hooks
-#   rev: master  # or specific git tag
-#   hooks:
-#     - id: markdownlint # Configure in .mdlrc
-#       files: "content/"
+      - id: markdownlint
+        name: Markdown Lint
+        description: Lint newly changed Markdown files
+        entry: mdl
+        language: ruby
+        types: ["markdown"]
+
+  - repo: https://github.com/tcort/markdown-link-check
+    rev: v3.10.3
+    hooks:
+      - id: markdown-link-check
+        name: Markdown Link Check
+        description: Check links in newly changed Markdown files
+        entry: markdown-link-check
+        args: ["--config", "mlc_config.json"]
+        language: node
+        types: ["markdown"]
```

Comment:
We shouldn't let external links block commits (or merges), they will fail sometimes due to websites being down, rate limiting, DOS protection.

If we can change this to _only_ check internal links that would be worthwhile, otherwise it can probably be dropped all together.

(From CI too probably. I've found [Lychee](https://lychee.cli.rs/) recently and much prefer it to `html-proofer` so should make that change at some point)

Reply:
I think, actually, it only checks external links. This is because almost all of our internal links are of the form

`{{< relref "/docs/onboarding/new_joiners/systems_set_up.md" >}}`

or

`{{< ref "/content/docs/contributing/contributing_changes.md#continuous-integration" >}}`

and it only looks for links which look like `[text](link)`.

So maybe just drop the whole thing?